rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1094,7 +1094,7 @@ stream {
             set $location_path  {{ $ing.Path | escapeLiteralDollar | quote }};
             set $global_rate_limit_exceeding n;
 
-            {{ buildOpentracingForLocation $all.Cfg.EnableOpentracing $location }}
+            {{ buildOpentracingForLocation $all.Cfg.EnableOpentracing $all.Cfg.OpentracingTrustIncomingSpan $location }}
```

Comment:
I think the main difference here is:
Opentracing can be enabled globally, while we can or cannot enable trusting incoming span locally right?

So my question is: what is the behavior, as the cluster admin we wanna do? Leave this decision to the user? If so, this should be exclusively local (and we don't need to assess this here).

If we want to have precedence, than we need to have a configmap that explicits if we trust incoming span or not, and bypass the verification of the location

Reply:
This PR adds that configmap value. Without this PR (current state), all incoming spans are accepted as default behaviour. The change introduced here is the option to set a global "off" switch for that. The possible scenarios are:
- Spans are trusted globally, but certain backend/ingress owners decide they don't want to trust any spans
- Spans are not trusted globally, but certain backend/ingress owners decide that they do want to trust any spans

Seeing as we don't know every user of ingress-nginx, I'd prefer to keep it flexible. However, if you feel that there is a principle to be followed (for example, when trust is globally disabled then no one backend owner should have the right to enable it for just them), I am willing to change this PR to become more opinionated.